pkg/ecdsa/dkg/marshaling_test.go

```diff
@@ -0,0 +1,219 @@
+package dkg
+
+import (
+	fuzz "github.com/google/gofuzz"
+	"github.com/keep-network/keep-core/pkg/crypto/ephemeral"
+	"github.com/keep-network/keep-core/pkg/internal/pbutils"
+	"github.com/keep-network/keep-core/pkg/protocol/group"
+	"reflect"
+	"testing"
+)
+
+func TestEphemeralPublicKeyMessage_MarshalingRoundtrip(t *testing.T) {
+	keyPair1, err := ephemeral.GenerateKeyPair()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	keyPair2, err := ephemeral.GenerateKeyPair()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	publicKeys := make(map[group.MemberIndex]*ephemeral.PublicKey)
+	publicKeys[group.MemberIndex(211)] = keyPair1.PublicKey
+	publicKeys[group.MemberIndex(19)] = keyPair2.PublicKey
+
+	msg := &ephemeralPublicKeyMessage{
+		senderID:            group.MemberIndex(38),
+		ephemeralPublicKeys: publicKeys,
+	}
+	unmarshaled := &ephemeralPublicKeyMessage{}
+
+	err = pbutils.RoundTrip(msg, unmarshaled)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if !reflect.DeepEqual(msg, unmarshaled) {
+		t.Fatalf("unexpected content of unmarshaled message")
+	}
+}
+
+func TestFuzzEphemeralPublicKeyMessage_MarshalingRoundtrip(t *testing.T) {
+	for i := 0; i < 10; i++ {
+		var (
+			senderID            group.MemberIndex
+			ephemeralPublicKeys map[group.MemberIndex]*ephemeral.PublicKey
+		)
+
+		f := fuzz.New().NilChance(0.1).
+			NumElements(0, 512).
+			Funcs(pbutils.FuzzFuncs()...)
```

Comment:
ah, so `gofuzz` is essentially doing generative testing for us. Seems useful for the retry logic testing too!